be/src/storage/memtable.cpp

```diff
@@ -356,14 +367,21 @@ Status MemTable::_split_upserts_deletes(ChunkPtr& src, ChunkPtr* upserts, std::u
     return Status::OK();
 }
 
-void MemTable::_sort_column_inc() {
+void MemTable::_sort_column_inc(bool by_sort_key) {
     Columns columns;
-    int sort_columns = _vectorized_schema->num_key_fields();
-    for (int i = 0; i < _vectorized_schema->num_key_fields(); i++) {
-        columns.push_back(_chunk->get_column_by_index(i));
+    std::vector<ColumnId> sort_key_idxes;
+    if (!by_sort_key) {
```

Comment:
You can define an alias in Schema when the sort_key_idxes is the same as the keys_idxes.

Reply:
if sort key is not ordered prefix of primary key, sort by primary key idxes firstly, then by sort key, so sort key idxes is not a alias of primary key idxes in this scenario.